tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs

```diff
@@ -450,8 +450,16 @@ internal static class Telemetry
             public const string Enabled = "DD_INSTRUMENTATION_TELEMETRY_ENABLED";
 
             /// <summary>
-            /// Configuration key for the telemetry URL where the Tracer sends telemetry.
-            /// Ignored (and <see cref="ExporterSettings.AgentUri"/> is used instead) unless <see cref="ConfigurationKeys.ApiKey"/> is set.
+            /// Configuration key for sending telemetry direct to telemetry intake. If enabled, and
+            /// <see cref="ConfigurationKeys.ApiKey"/> is set, sends telemetry direct to intake. Otherwise, sends
+            /// telemetry to Agent. Enabled by default if <see cref="ConfigurationKeys.ApiKey"/> is available.
+            /// </summary>
+            public const string AgentlessEnabled = "DD_TRACE_TELEMETRY_AGENTLESS_ENABLED";
```

Comment:
I was going to say, should this setting use the `DD_INSTRUMENTATION_TELEMETRY_*` prefix like `DD_INSTRUMENTATION_TELEMETRY_ENABLED`? But then I noticed `DD_TRACE_TELEMETRY_URL` below. 😕

Reply:
> I wish we didn't have this inconsistency.

Yeah... that's on me 😬 The `DD_INSTRUMENTATION_TELEMETRY_ENABLED` key was the prescribed one, whereas the others are keys I added because I needed them. 

I'm totally ok with removing the `TRACE_` from these (I'm not worried about the breaking change aspect), but... should we? Then we're going against the `DD_TRACE_` standard 🤷‍♂️ But then we're not _just_ tracing now so... 😄

Reply:
I think it's ok to drop the `TRACE` since these are not strictly related to tracing. They also affect AppSec, for example, and in theory the dogstats client could use the same setting, too.  See also: `DD_RUNTIME_METRICS_ENABLED` for another non-tracing setting.

Reply:
I'll go for that then. I also think it's ok to remove the `TRACE_` from the existing `DD_TRACE_TELEMETRY_URL`. Breaking change technically, but _very_ unlikely to affect anyone IMO